src/ex2.ml

```diff
@@ -0,0 +1,35 @@
+let loadXMLDoc () =
+  let xmlhttp = XmlHttpRequest.create () in
+  let onreadystatechange () =
+    match xmlhttp##readyState, xmlhttp##status with
+    | XmlHttpRequest.DONE, 200 ->
+      let xmlDoc = xmlhttp##responseXML in
+      let buf = Buffer.create 16 in
+      begin match Js.Opt.to_option xmlDoc with
+        | None -> ()
+        | Some xmlDoc ->
+          let elts = xmlDoc##getElementsByTagName (Js.string "ARTIST") in
+          for i = 0 to elts##length - 1 do
+            match Js.Opt.to_option (elts##item (i)) with
+            | None -> ()
+            | Some elt ->
+              begin match Js.Opt.to_option elt##firstChild with
+                | None -> ()
+                | Some elt ->
+                  begin match Js.Opt.to_option elt##nodeValue with
+                    | None -> ()
+                    | Some elt ->
+                      Buffer.add_string buf (Js.to_string elt);
+                      Buffer.add_string buf "<br>"
```

Comment:
I appreciate your comments. It helps me discover js_of_ocaml libraries and write better code.